src/lsp/cobol_typeck/old_env_builder.ml

```diff
@@ -98,7 +98,8 @@ let for_compilation_unit =
       let name = Cobol_ptree.program_name' name in
       acc_result @@ fun parent_env -> build_env name env ?parent_env
     method! fold_function_unit f =
-      acc_result @@ fun _ -> build_env f.function_name f.function_env
+      acc_result @@ fun _ -> build_env
```

Comment:
I'd suggest to take inspiration from `Cobol_ptree.program_name'` and maybe have a `Cobol_ptree.function_name'`.